File: Week6/sentimental-cash/cash.py
from cs50 import get_float

while True:
    change = get_float("Change: ")
    if change > 0:
        break

n = 0

while change > 0.249:
    change -= 0.25
    n += 1
    # print(change)
while change > 0.09:
    change -= 0.1
    n += 1
    # print(change)
while change > 0.049:
    change -= 0.05
    n += 1
    # print(change)
while change > 0.009:
    change -= 0.01
    n += 1
    # print(change)

print(n)

File: Week1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int select_max(int remainder);

int main(void)
{
    // 接受非负输入：Change owed
    int owed;
    do
    {
        owed = get_int("Change owed: ");
    }
    while (owed < 0);

    // 贪心选择最大可支付硬币, n 为选择次数（也是硬币数量）
    int n = select_max(owed);
    printf("%d\n", n);
}

// 全局变量记录选择/递归次数
int recursion_count = 0;

int select_max(int remainder)
{

    // 递归边界
    if (remainder == 0)
    {
        return recursion_count;
    }

    // 递归体, 从大到小选择
    if (remainder >= 25)
    {
        remainder -= 25;
        recursion_count++;
        return select_max(remainder);
    }
    else if (remainder >= 10 && remainder < 25)
    {
        remainder -= 10;
        recursion_count++;
        return select_max(remainder);
    }
    else if (remainder >= 5 && remainder < 10)
    {
        remainder -= 5;
        recursion_count++;
        return select_max(remainder);
    }
    else
    {
        remainder -= 1;
        recursion_count++;
        return select_max(remainder);
    }
}
